### PR TITLE
Add dust history

### PR DESCRIPTION
**Pull request by MatthieuSchaller**
Basically a copy of what is done for the stellar masses. Makes use of the new column we added to the stats file to record the total dust mass of the universe.

### CODE REVIEW

File: colibre/config.yml [hunk 1]
```diff
@@ -508,3 +508,8 @@ scripts:
     caption: SNIa rate history plotted directly from the SNIa.txt file produced by SWIFT.
     section: SNIa Rate
     output_file: SNIa_rate_history.png
+  - filename: scripts/dust_mass_evolution.py
+    caption: Evolution of the dust mass density plotted directly from the statistics.txt file produced by SWIFT.
+    output_file: dust_mass_evolution.png
+    section: Star Formation History
```
Comment (EvgeniiChaikin) [inline colibre/config.yml:514]: Is this the correct section that we want for this plot?

File: colibre/scripts/dust_mass_evolution.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
+"""
+Plots the dust mass density evolution.
+"""
+import matplotlib
+
+import unyt
```
Comment (EvgeniiChaikin) [inline colibre/scripts/dust_mass_evolution.py:6]:
```suggestion
```

File: colibre/scripts/dust_mass_evolution.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+import matplotlib.pyplot as plt
+import numpy as np
+import sys
```
Comment (EvgeniiChaikin) [inline colibre/scripts/dust_mass_evolution.py:10]:
```suggestion
```

File: colibre/scripts/dust_mass_evolution.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+# Observational data plotting
+
+observational_data = glob.glob(
+    f"{arguments.config.config_directory}/{arguments.config.observational_data_directory}/data/DustMassDensity/*.hdf5"
+)
+
+for index, observation in enumerate(observational_data):
+    obs = load_observation(observation)
+    obs.plot_on_axes(ax)
```
Comment (EvgeniiChaikin) [inline colibre/scripts/dust_mass_evolution.py:72]:
```suggestion
path_to_obs_data = f"{arguments.config.config_directory}/{arguments.config.observational_data_directory}"
observational_data = load_observations(
    sorted(glob.glob(f"{path_to_obs_data}/data/DustMassDensity/*.hdf5"))
)

for obs_data in observational_data:
    obs_data.plot_on_axes(ax)
```

File: colibre/scripts/dust_mass_evolution.py [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+import numpy as np
+import sys
+import glob
+
+
+from swiftsimio import load, load_statistics
+
+from swiftpipeline.argumentparser import ScriptArgumentParser
+
+from velociraptor.observations import load_observation
```
Comment (EvgeniiChaikin) [inline colibre/scripts/dust_mass_evolution.py:18]:
```suggestion
from velociraptor.observations import load_observations
```

- `load_observation` is outdated. 
- Note also the associated suggestion below in this file where I use `load_observations`

File: colibre/scripts/dust_mass_evolution.py [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+    "$100$",
+]
+a_ticks = 1.0 / (redshift_ticks + 1.0)
+
+ax.set_xticks(a_ticks)
+ax.set_xticklabels(redshift_labels)
+ax.tick_params(axis="x", which="minor", bottom=False)
+
+ax.set_xlim(1.02, 0.07)
+ax.set_ylim(1e6, 1.3e9)
```
Comment (EvgeniiChaikin) [inline colibre/scripts/dust_mass_evolution.py:98]: In my tests of this branch, I needed to lower the lower limit to 1e4 in order to see dust in the plot.
```suggestion
ax.set_ylim(1e4, 1.3e9)
```

**Review by EvgeniiChaikin — CHANGES_REQUESTED** [review]
Thank you @MatthieuSchaller!

I added a few small comments to improve the file.

**One major comment**. 

There is a bug on the `swift` side, in https://gitlab.cosma.dur.ac.uk/EAGLE/swift-colibre/-/blob/master/src/statistics.c#L900

The length of the `Cross Helicity ` string is 14, which is as large as that in the format specifier. Because of this, in the output of `statistics.txt`, there is only one space between `Cross Helicity` and the previous field, which means that when you read `statistics.txt` through `swiftsimio`, `swiftsimio` reads `Cross Helicity` and the name of the previous field as a _single_ string. Hence, values of all fields stored after `Cross Helicity` in `statistics.txt` will be bound to incorrect columns in the statistics file (shifted by 1 from the correct ones). Since `dust_mass` is currently stored last in `statistics.txt`, values of `dust_mass` are also bound to an incorrect column.

https://gitlab.cosma.dur.ac.uk/EAGLE/swift-colibre/-/blob/master/src/statistics.c#L900

**Comment by MatthieuSchaller** [comment]
I'll fix the swift thing. 

For the other suggestions then we should change all the related plots for consistency.

File: colibre/config.yml (hunk 1 above)
Comment (MatthieuSchaller) [reply colibre/config.yml:514]: I thought next to the SF and stellar mass density was good.

Comment (MatthieuSchaller) [reply colibre/config.yml:514]: Feel free to change it to something you prefer.

**Comment by MatthieuSchaller** [comment]
I have fixed things on the SWIFT side. 

For this MR, just editing the header will work I presume.

**Comment by MatthieuSchaller** [comment]
I agree with all the technical suggestions. 

Back to you. :)

File: colibre/config.yml (hunk 1 above)
Comment (EvgeniiChaikin) [inline colibre/config.yml:514]:
```suggestion
    section: Dust Evolution
```

**Review by EvgeniiChaikin — APPROVED** [review]

**Comment by EvgeniiChaikin** [comment]
I added one last comment with a suggestion to change the section where the plot will be shown. Otherwise, this PR looks ready!

File: colibre/config.yml (hunk 1 above)
Comment (MatthieuSchaller) [reply colibre/config.yml:514]: Ok, good if you prefer a new section.

**Pull request merged**